File: server/webapp/WEB-INF/rails.new/app/controllers/api_v1/admin/encryption_controller.rb
##########################################################################
# Copyright 2016 ThoughtWorks, Inc.
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
##########################################################################

module ApiV1
  module Admin
    class EncryptionController < ApiV1::BaseController
      before_action :check_admin_user_or_group_admin_user_and_401

      @@go_cipher = GoCipher.new

      def encrypt_value
        render DEFAULT_FORMAT => {encrypted_value: @@go_cipher.encrypt(params[:value])}
      rescue
        render_message("An error occurred while encrypting the value. Please check the logs for more details.", :internal_server_error)
      end
    end
  end
end

File: api/api-encryption-v1/src/main/java/com/thoughtworks/go/apiv1/admin/encryption/spring/EncryptionController.java
/*
 * Copyright 2021 ThoughtWorks, Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package com.thoughtworks.go.apiv1.admin.encryption.spring;

import com.google.common.base.Ticker;
import com.thoughtworks.go.api.spring.ApiAuthenticationHelper;
import com.thoughtworks.go.apiv1.admin.encryption.EncryptionControllerDelegate;
import com.thoughtworks.go.security.GoCipher;
import com.thoughtworks.go.spark.spring.SparkSpringController;
import com.thoughtworks.go.util.SystemEnvironment;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;

@Component
public class EncryptionController implements SparkSpringController {
    private static final int DEFAULT_REQUESTS_PER_MINUTE = SystemEnvironment.getMaxEncryptionAPIRequestsPerMinute();
    private final EncryptionControllerDelegate delegate;

    @Autowired
    public EncryptionController(ApiAuthenticationHelper apiAuthenticationHelper) {
        delegate = new EncryptionControllerDelegate(apiAuthenticationHelper, new GoCipher(), DEFAULT_REQUESTS_PER_MINUTE, Ticker.systemTicker());
    }

    @Override
    public void setupRoutes() {
        delegate.setupRoutes();
    }
}
